docs/sdk/src/guides/your_first_runtime.rs

```diff
@@ -1,3 +1,169 @@
+//! This guide will walk you through the steps to add your pallet to a runtime.
+//!
+//! The good news is, in [`crate::guides::your_first_pallet`], we have already created a _test_
+//! runtime that was used for testing, and a real runtime is not that much different!
+//!
+//! ## Setup
+//!
+//! A runtime shares a few similar setup requirements as with a pallet:
+//!
+//! * importing [`frame`], [`codec`], and [`scale_info`] crates.
+//! * following the [`std`](crate::polkadot_sdk::substrate#wasm-build) pattern.
+//!
+//! But, more specifically, it also contains:
+//!
+//! * a `build.rs` that uses [`substrate_wasm_builder`]. This entails declaring
+//!   `[build-dependencies]` in the Cargo manifest file:
+//!
+//! ```ignore
+//! [build-dependencies]
+//! substrate-wasm-builder = { ... }
+//! ```
+//!
+//! >Note that a runtime must always be one-runtime-per-crate. You cannot define multiple runtimes
+//! per rust crate.
+//!
+//! You can find the full code of this guide in [`first_runtime`].
+//!
+//! ## Your First Runtime
+//!
+//! The first new property of a real runtime that it must define its
+//! [`frame::runtime::prelude::RuntimeVersion`]:
+#![doc = docify::embed!("./packages/guides/first-runtime/src/lib.rs", VERSION)]
+//!
+//! The version contains a number of very important fields, such as `spec_version` and `spec_name`
+//! that play an important role in identifying your runtime and its version. More about runtime
+//! upgrades in [`crate::reference_docs::frame_runtime_upgrades_and_migrations`].
+//!
+//! Then, a real runtime also contains the `impl` of all individual pallets' `trait Config` for
+//! `struct Runtime`, and a [`frame::runtime::prelude::construct_runtime`] macro that amalgamates
+//! them all.
+//!
+//! In the case of our example:
+#![doc = docify::embed!("./packages/guides/first-runtime/src/lib.rs", our_config_impl)]
+//!
+//! In this example, we bring in a number of other pallets from [`frame`] into the runtime, each of
+//! their `Config` need to be implemented for `struct Runtime`:
+#![doc = docify::embed!("./packages/guides/first-runtime/src/lib.rs", config_impls)]
+//!
+//! Notice how we use [`frame::pallet_macros::derive_impl`] to provide "default" configuration items
+//! for each pallet. Feel free to dive into the definition of each default prelude (eg.
+//! [`frame::prelude::frame_system::pallet::config_preludes`]) to learn more which types are exactly
+//! used.
+//!
+//! Recall that in [`crate::guides::your_first_pallet`], we provided `type AccountId = u64` to
+//! `frame_system`, while in this case we rely on whatever is provided by
+//! [`SolochainDefaultConfig`], which is indeed a "real" 32 byte account id.
+//!
+//! Then, a familiar instance of `construct_runtime` amalgamates all of the pallets:
+#![doc = docify::embed!("./packages/guides/first-runtime/src/lib.rs", cr)]
+//!
+//! Recall from [`crate::reference_docs::wasm_meta_protocol`] that every (real) runtime needs to
```

Comment:
nit: I would shorten all the links to sections/chapters, (could be done in some follow-up) - probably wold be easier to read.
```suggestion
//! Recall from [`wasm_meta_protocol`](`crate::reference_docs::wasm_meta_protocol`) that every (real) runtime needs to
```

Reply:
I would like to write a script with ChatGPT that grabs all links, and automatically moved them all to the bottom of the file. This is best both for authors and readers.

Reply:
https://github.com/paritytech/polkadot-sdk/issues/6195